frontend/src/App.tsx

```diff
@@ -1,19 +1,72 @@
-import './App.css';
-import Ready from './pages/ReadyPage/Ready';
+import { useState, useEffect } from 'react';
+import axios from 'axios';
+import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
 import CharacterSelection from './pages/CharacterSelectionPage/CharacterSelection';
+import Ready from './pages/ReadyPage/Ready';
 import StartPage from './pages/StartPage/StartPage';
 import SceneOnePage from './pages/SceneOnePage/SceneOnePage';
 import SceneTwoPage from './pages/SceneTwoPage/SceneTwoPage';
 import SceneThreePage from './pages/SceneThreePage/SceneThreePage';
-import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
-import SceneGoalPage from "./pages/SceneGoalPage/SceneGoalPage.tsx";
+import SceneGoalPage from "./pages/SceneGoalPage/SceneGoalPage";
+import { HeroDto, SaveGameDto } from "./types/types.ts";
 
 export default function App() {
+    const [saveGames, setSaveGames] = useState<SaveGameDto[]>([]);
+    const [heroes, setHeroes] = useState<HeroDto[]>([]);
+    const [error, setError] = useState<string | null>(null);
+    const [shouldUpdate, setShouldUpdate] = useState(false);
+
+    useEffect(() => {
+        const fetchDataAndUpdate = async () => {
+            try {
+                const [saveGamesResponse, heroesResponse] = await Promise.all([
+                    axios.get<SaveGameDto[]>('/api/savegames'),
+                    axios.get<HeroDto[]>('/api/heroes')
+                ]);
```

Comment:
Why are you using `Promise.all` here?

Reply:
Hey Martin, great question! I used Promise.all for efficiency. It lets us fetch save games and heroes data at the same time, and it's neater for handling any errors that come up. Quick and clean!